tests/test_graph.py

```diff
@@ -52,78 +52,78 @@ def bad_tree():
          t
     """
     t = OvalNode(
-        1, "value", "true", False, None, None,  [
+        1, "value", "true", False, None, None, None, [
```

Comment:
General remark: You seem to have to update tests quite a lot if you make a change in API. In cases like this, it would be worth thinking to use named arguments and arguments with default values, so you don't have to alter all those function invocations except cases when you want to pass something else than `None`.
Named arguments help against confusion what has been passed and what is supplied by default.

Reply:
I'll do it in another PR.